src/builder.rs

```diff
@@ -228,17 +239,105 @@ where
-    pub fn connect_tagged_builder_signals<T, C, AA>(&self, ctx: &mut C, tag: T) -> &Self
+    pub fn connect_signals_tagged<A, S, T>(&mut self, tag: T, ctx: &mut actix::Context<A>)
     where
         T: Clone + 'static,
-        AA: actix::Actor<Context = C>,
-        C: actix::AsyncContext<AA>,
-        AA: actix::StreamHandler<(T, S)>
+        A: actix::Actor<Context = actix::Context<A>>,
+        A: actix::StreamHandler<(T, S)>,
+        S: BuilderSignal,
     {
-        if let Some(rx) = self.stream_builder_signals() {
-            let stream = rx.map(move |s| (tag.clone(), s));
-            ctx.add_stream(stream);
+        let (tx, rx) = mpsc::channel(16);
+        let rx = rx.map(move |s| (tag.clone(), s));
+        for signal in S::list_signals() {
+            S::bridge_signal(signal, tx.clone());
         }
-        self
+        use actix::AsyncContext;
+        ctx.add_stream(rx);
+    }
+
+    /// Create an Actix actor and connect it to the builder's widgets and signals.
+    ///
+    /// `make_actor` is a function that receives the actor context and the widgets, and is
+    /// responsible for constructing the actor struct with the widgets inside it. It can also be
+    /// used for configuring and or showing the widgets GTK-wise (though this can also be handled
+    /// by the actor afterwards)
+    pub fn new_actor<A, F>(&mut self, make_actor: F) -> actix::Addr<A>
+    where
+        A: actix::Actor<Context = actix::Context<A>>,
+        F: FnOnce(&mut ActorBuilderContext<A>) -> A,
+    {
+        <A as actix::Actor>::create(move |ctx| {
+            let mut ctx = ActorBuilderContext {
+                builder: self,
+                actor: std::marker::PhantomData,
+                ctx,
+            };
+            make_actor(&mut ctx)
+        })
+    }
+
+    /// Create a stream of all the signals.
+    ///
+    /// Will return `None` if there are no signals, to allow avoiding closed streams.
+    pub fn finish(self) {
+        std::mem::drop(self)
     }
 }
+
+impl Drop for BuilderConnector {
+    fn drop(&mut self) {
+        use gtk::prelude::BuilderExtManual;
+
+        let callbacks = &mut self.callbacks;
+        self.builder.connect_signals(move |_, signal| {
+            callbacks.remove(signal).unwrap_or_else(|| Box::new(|_| None))
+        });
+    }
+}
+
+/// Type of a gtk signal callback function that operates on uncast glib values
+pub type RawSignalCallback = Box<dyn Fn(&[glib::Value]) -> Option<glib::Value>>;
+
+pub fn make_signal_handler<A, S>(
+    handler_name: &str,
+    ctx: &mut A::Context,
+) -> RawSignalCallback 
+where
+    A: actix::Actor<Context = actix::Context<A>>,
+    A: actix::StreamHandler<S>,
+    S: BuilderSignal,
+{
+    let (tx, rx) = mpsc::channel(16);
+    A::add_stream(rx, ctx);
+    S::bridge_signal(handler_name, tx)
+        .ok_or_else(|| format!("Handler '{}' was requested, but only {:?} exist", handler_name, S::list_signals()))
```

Comment:
Probably not a good idea to render a list of **all** the signals. We don't know how long it'll be. It's better to use [`std::any::type_name<S>`](https://doc.rust-lang.org/std/any/fn.type_name.html).

Reply:
I added it because it was useful for debugging. But maybe we cut it if in the case it exceeds a specific length?